src/lib/core/cube.tsx

```diff
@@ -0,0 +1,9 @@
+import { CubeIndex } from "./cubeindex"
+import { CubeOp } from "./cubeop"
+
+export interface Cube {
+  readonly dimension: number;
+  readonly positionOf: (facelet: number) => CubeIndex;
+  readonly occupantOf: (position: CubeIndex) => number;
```

Comment:
idk if you want to add more context to these primitives (e.g. `number`). But if so, you can make type aliases:
```
type FaceIndex = number;
type Color = "red" | "white" | "blue" | "green" | "orange" | "yellow"
```

then reference the alias instead of `number`

Reply:
I think he might want to abstract way from specific colors and just be able to use facelet indicies, in which case I don't think there's a simple way where we can enforce that type

Reply:
yeah, the color was just meant to show what you can do with types. The `Facelet = number` is the more relevant example.

Reply:
That's neat. Done.